backend/server/public/src/components/addetail.jsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import { connect } from "react-redux";
+import Ad from "components/ad.jsx";
+import { withRouter, Link } from "react-router-dom";
+import { getOneAd } from "../actions.js";
+
+export class AdDetailUnconnected extends React.Component {
+  componentDidMount() {
+    // if this.props.ad, we're good, we got the ad prop from <Frontend> or it's already been fetched.
+    // if !this.props.ad && !this.props.getOneAd, then something has gone wrong.
+    if (
+      !(this.props.ad && this.props.ad.id) &&
+      this.props.getOneAd &&
```

Comment:
What's `getOneAd`? This seems really complicated.

Reply:
It's the function that goes to the Rust API to get the data for the permalinked ad.